### PR TITLE
Add useful fields to Firebase project resource.

### PR DESCRIPTION
**Pull request by mraouffouad**
Adding projectId, resources and annotations fields to Firebase project. Also, modifying the type of projectNumber field to Integer.

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement

```

### CODE REVIEW

File: mmv1/products/firebase/api.yaml
Comment (tylerg-dev) [inline mmv1/products/firebase/api.yaml:67]: it would be uncommon for a developer to have a FirebaseProject in their .tf file and not have the parent Cloud Project.  I suspect we should omit Project Id as [this](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_project#project_id) is the preferred place to read it.

File: mmv1/products/firebase/api.yaml
Comment (tylerg-dev) [inline mmv1/products/firebase/api.yaml:87]: default resources are nearly obsolete.  Lets omit this one unless there is a specific customer issue asking for it.

If we do keep this, our description will need to explicitly call out this is going to be deprecated.

File: mmv1/products/firebase/api.yaml
Comment (tylerg-dev) [inline mmv1/products/firebase/api.yaml:132]: project location is handled its own resource right now: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/firebase_project_location

File: mmv1/products/firebase/api.yaml
Comment (tylerg-dev) [inline mmv1/products/firebase/api.yaml:216]: was this intended to be in this pull request?

**Review by tylerg-dev — COMMENTED** [review]
Can you squash all of your commits together into a single overall commit.  The upstream maintainers will prefer to have the merge be a single commit in the mainline history.

File: mmv1/products/firebase/api.yaml
Comment (mraouffouad) [reply mmv1/products/firebase/api.yaml]: No, I mistakenly branched this off the first PR branch instead of main.

File: mmv1/products/firebase/api.yaml
Comment (mraouffouad) [reply mmv1/products/firebase/api.yaml:67]: Done.

File: mmv1/products/firebase/api.yaml
Comment (mraouffouad) [reply mmv1/products/firebase/api.yaml:132]: Are you suggesting to remove it from here?

Comment (tylerg-dev) [reply mmv1/products/firebase/api.yaml:132]: probably.  it's nearly an obsolete concept, and it would create redundant ways to interact with it.  

If we omit it, we can just retire/deprecate the old project_location resource later when it is no longer meaningful.

**Comment by shuyama1** [comment]
Looks like the failing tests `testAccFirebaseProject_firebaseProjectBasicExample` is related to the change of this PR. Please let me know if you need help debugging!

File: mmv1/products/firebase/api.yaml
Comment (mraouffouad) [reply mmv1/products/firebase/api.yaml:132]: I will do this in a separate PR.

**Comment by mraouffouad** [comment]
@shuyama1 Thanks for reviewing. All new fields are output-only except for "annotations". As for the failing test, it's complaining that this field (annotation) is not found. While it exists in Firebase project proto in beta, it's not being populated in the request of AddFirebase API. Could this be the reason? Should I remove it from the .tf.erb file? I suspect this would fix the test, but how would we be able to use later with other APIs when they get added to the api.yaml later on?

File: mmv1/products/firebase/api.yaml
Comment (mraouffouad) [reply mmv1/products/firebase/api.yaml:81]: i removed this field as it's not included in the addFirebase API.

**Comment by shuyama1** [comment]
@mraouffouad Thanks for update the PR. Is this ready to review? If so, The question related to the `projecId` field is still unsolved.

**Comment by mraouffouad** [comment]
> @mraouffouad Thanks for updating the PR. Is this ready to review? If so, The problem related to the `projecId` field is still unsolved.

@shuyama1 Yes, the PR is ready for review.

I had responded to your comment regarding projectId, it's the same as project. This field is removed now. Thanks.

File: mmv1/products/firebase/api.yaml
Comment (shuyama1) [reply mmv1/products/firebase/api.yaml:73]: I am still seeing the `annotations` in the latest change. Just want to make sure if it's intentional.

**Review by shuyama1 — COMMENTED** [review]
Based on the previous discussion, it sounds like the `annotations` is not supported in the public API yet. I just want to make sure if that statement still holds. If so, let's remove the field from the resource schema. (I've seen it's removed from the test but we also need to exclude is from the resource as currently setting the field is not supported if it's not released in the API).

**Comment by mraouffouad** [comment]
> 
The field is released to public already. We're asking devrel to regenerate the public docs, which should include that field soon.

**Comment by shuyama1** [comment]
@mraouffouad Thanks for the update! Let take back the test on `annotations` then.

**Comment by mraouffouad** [comment]
Hmm, that test would still fail if we add ```annotations``` back regardless whether it's public in not. The addFirebase API that we use in create_url in api.yaml doesn't set this field (hence it's not part of the addFirebase request proto). It's just add our master service account on the project and enables Firebase API. That field could be set in other APIs, such as updateFirebaseProject.

**Comment by shuyama1** [comment]
@mraouffouad I am a bit confused here. Sorry if I misunderstood it. If the API of this resource does not support this `annotations` field, then I think we should remove it from the schema.

**Comment by tylerg-dev** [comment]
The resource does support annotations and the field is publicly accessible since August I think.  We just have not managed to get the publicly published Docs updated yet (DevRel time has been quite constrained due to Firebase's yearly conference in October).

**Comment by mraouffouad** [comment]
> The resource does support annotations and the field is publicly accessible since August I think. We just have not managed to get the publicly published Docs updated yet (DevRel time has been quite constrained due to Firebase's yearly conference in October).

**Comment by mraouffouad** [comment]
@shuyama1  is there anything blocking this PR at this point?

**Comment by mraouffouad** [comment]
> shuyama1

@shuyama1 fair. I removed the annotations field for now.

**Review by shuyama1 — COMMENTED** [review]
LGTM in general. Just some nit picks.

**Review by shuyama1 — CHANGES_REQUESTED** [review]
@mraouffouad There's still some comments that are addressed by the latest update.  Would you mind take a look at my last comment? Plus can you update the PR description as I don't think it's accurate after the new updates.

**Pull request closed without merging**
